### PR TITLE
OCPQE-15774: Fix pruning

### PR DESCRIPTION
**Pull request by aleskandro**
Using the post steps defined in openshift/release will allow to
always execute the most up-to-date post chain and avoid duplicating the
deprovisioning code.
This commit introduces:
- a deprovision.sh script in the prow-image: the post chain in
  openshift/release are not expected to be concurrent safe and executing
  them in the prow-image will ensure a similar environment to the one in
  the post chain pods
- changes the previous implementation of the prune_nodes script to run a
  container and the deprovision.sh script in the prow-image
- patch the clean-up.sh script to run based on the /var/builds/*
  content
- (vendor) enable the systemd timer unit

Finally, [OCPQE-18449](https://issues.redhat.com//browse/OCPQE-18449) is resolved by checking the existence of the file /var/builds/$cluster_name/preserve

Testing of this PR is done piece by piece:

1. the deprovision.sh script in the prow-image is tested through an additional volume
2. the prune_nodes changes in the fcos-bastion-image are tested locally by executing it
3. the timer unit's script changes are tested locally too.

### CODE REVIEW

File: images/fcos-bastion-image/root/usr/bin/prune_nodes [hunk 1]
```diff
@@ -1,216 +1,11 @@
...
-
-# Restart dnsmasq container
-docker restart dhcpd
-
-# Reload and flush bind9 container
-docker exec bind9 rndc reload && \
-  docker exec bind9 rndc flush
-
-echo "<4>$CLUSTER - Clean-up, wipe disks and deprovisioning completed successfully"
+podman run -it -v "/root/.ssh/id_rsa:/var/builds/$1/ssh-key:z" -v "/var/builds/:/var/builds/:Z" \
```
Comment (sgoveas) [inline images/fcos-bastion-image/root/usr/bin/prune_nodes:8]: Would it be better for the second volume be restricted to the respective container?
```-v "/var/builds/$1:/var/builds/$1:Z"```

Comment (aleskandro) [reply images/fcos-bastion-image/root/usr/bin/prune_nodes:8]: As that folder is going to be deleted before the container quits, I wanted to avoid it.

**Pull request merged**